network/codec_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2017 The Alvalor Authors
+//
+// This file is part of Alvalor.
+//
+// Alvalor is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Alvalor is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with Alvalor.  If not, see <http://www.gnu.org/licenses/>.
+
+package network
+
+import (
+    "testing"
+    "bytes"
+    "math/rand"
+    "github.com/stretchr/testify/assert"
+)
+
+func TestEncodeUnknownType(t *testing.T) {
+    codec := SimpleCodec{}
+
+    buf := &bytes.Buffer{}   
+    msg := 1
+    err := codec.Encode(buf, msg)
+
+    assert.NotNil(t, err, "Expected to receive error in case message type is unknown")
+}
+
+func TestDecodeUnknownCode(t *testing.T) {
+    codec := SimpleCodec{}
+
+    buf := make([]byte, 1)
```

Comment:
I would not call `buf`, which is usually `bytes.Buffer` the same as the byte slice.
Usually, I would use `data` or `bytes` to make the distinction.